precepte-core-cats/src/main/scala/package.scala

```diff
@@ -77,26 +129,34 @@ package object corecats extends SubMeta {
   //   def lift[F[_], A](f: F[StreamingT[A]]): StreamingT[F, A] = StreamingT.wait(f)
   // }
 
-  implicit def xorHasHoist[A]: HasHoist.Aux[({ type λ[α] = Either[A, α] })#λ, ({ type λ[F[_], B] = EitherT[F, A, B] })#λ] = new XorHasHoist[A]
-
-  def future[Ta, M, U, A](ta: Ta)(λ: Precepte[Ta, M, U, Future, A]#S => Future[A])(implicit func: Functor[Future], ec: scala.concurrent.ExecutionContext): Precepte[Ta, M, U, Future, Either[Throwable, A]] =
-    Precepte(ta){ pa =>
-      func.map(λ(pa))(Right(_))
-        .recover{ case e => Left(e) }
+  implicit def xorHasHoist[A]
+    : HasHoist.Aux[Either[A, ?], λ[(α[_], β) => EitherT[α, A, β]]] =
+    new XorHasHoist[A]
+
+  @SuppressWarnings(Array("org.wartremover.warts.ImplicitParameter"))
+  def future[Ta, M, U, A](ta: Ta)(λ: PState[Ta, M, U] => Future[A])(
```

Comment:
I know this wasn't modified by this PR but IMO either use `tag: Tag` or stick with `t: T` like all the others `Precepte[T, M, U, F, A]`